### PR TITLE
feat: Add first time interaction warning

### PR DESCRIPTION
**Pull request by OGPoyraz**
## **Description**

This PR implements first time interaction feature where it shows an alert if you interact with the address for the first time.
Information of the first time interaction is fetched in the transaction controller when the transaction is added to the state. 

Core PR: https://github.com/MetaMask/core/pull/4895

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/28435?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/3040

## **Manual testing steps**

1. Go to test dapp
2. Use send legacy transaction - make sure you interact here with your account for the first time
3. See warning

### **After**

Screenshot 2024-11-13 at 11 51 14

Screenshot 2024-11-13 at 11 51 19

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by matthewwalsh0 — DISMISSED** [review]

File: ui/pages/confirmations/hooks/alerts/transactions/useFirstTimeInteractionAlert.test.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  it('returns no alerts if no transactions', () => {
+    expect(
+      runHook({
+        currentConfirmation: CONFIRMATION_MOCK,
+        transactions: [],
+      }),
+    ).toEqual([]);
+  });
+
+  it('returns no alerts if firstTimeInteraction is false', () => {
```
[the remark this answers is not in the document]
Comment (OGPoyraz) [reply ui/pages/confirmations/hooks/alerts/transactions/useFirstTimeInteractionAlert.test.ts:83]: Done

File: app/_locales/en/messages.json [hunk 2]
```diff
@@ -2091,6 +2091,12 @@
     "message": "File import not working? Click here!",
     "description": "Helps user import their account from a JSON file"
   },
+  "firstTimeInteractionReason": {
```
[the remark this answers is not in the document]
Comment (OGPoyraz) [reply app/_locales/en/messages.json]: Thanks for the heads up

File: ui/pages/confirmations/hooks/alerts/transactions/useFirstTimeInteractionAlert.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import { useI18nContext } from '../../../../../hooks/useI18nContext';
+import { Severity } from '../../../../../helpers/constants/design-system';
+import { RowAlertKey } from '../../../../../components/app/confirm/info/row/constants';
+import { useConfirmContext } from '../../../context/confirm';
+
+export function useFirstTimeInteractionAlert(): Alert[] {
+  const t = useI18nContext();
+  const { currentConfirmation } = useConfirmContext();
+
+  const { isFirstTimeInteraction } = currentConfirmation as TransactionMeta;
```
[the remark this answers is not in the document]
Comment (OGPoyraz) [reply ui/pages/confirmations/hooks/alerts/transactions/useFirstTimeInteractionAlert.ts]: Done

File: ui/pages/confirmations/components/confirm/info/token-transfer/transaction-flow-section.tsx [hunk 4]
```diff
@@ -50,25 +56,52 @@ export const TransactionFlowSection = () => {
...
+                label={t('to')}
+                ownerId={transactionMeta.id}
+                style={{
+                  padding: 0,
+                  margin: 0,
+                  marginBottom: 4,
+                }}
+              >
+                {/* Intentional fragment */}
+                <></>
```
[the remark this answers is not in the document]
Comment (OGPoyraz) [reply ui/pages/confirmations/components/confirm/info/token-transfer/transaction-flow-section.tsx]: Done

File: app/scripts/metamask-controller.js [hunk 5]
```diff
@@ -1930,6 +1930,8 @@ export default class MetamaskController extends EventEmitter {
         updateTransactions: false,
       },
       isMultichainEnabled: process.env.TRANSACTION_MULTICHAIN,
+      isFirstTimeInteractionEnabled: () =>
```
[the remark this answers is not in the document]
Comment (OGPoyraz) [reply app/scripts/metamask-controller.js:1944]: Done

File: package.json [hunk 6]
```diff
@@ -256,7 +256,8 @@
     "@ledgerhq/evm-tools/axios": "^0.28.0",
     "@ledgerhq/hw-app-eth/axios": "^0.28.0",
     "@ledgerhq/hw-app-eth@npm:^6.39.0": "patch:@ledgerhq/hw-app-eth@npm%3A6.39.0#~/.yarn/patches/@ledgerhq-hw-app-eth-npm-6.39.0-866309bbbe.patch",
-    "@ledgerhq/evm-tools@npm:^1.2.3": "patch:@ledgerhq/evm-tools@npm%3A1.2.3#~/.yarn/patches/@ledgerhq-evm-tools-npm-1.2.3-414f44baa9.patch"
+    "@ledgerhq/evm-tools@npm:^1.2.3": "patch:@ledgerhq/evm-tools@npm%3A1.2.3#~/.yarn/patches/@ledgerhq-evm-tools-npm-1.2.3-414f44baa9.patch",
+    "@metamask/transaction-controller@^38.3.0": "npm:@metamask-previews/transaction-controller@39.0.0-preview-fefbd325"
```
[the remark this answers is not in the document]
Comment (OGPoyraz) [reply package.json]: Removed

File: ui/pages/confirmations/components/confirm/info/token-transfer/transaction-flow-section.tsx [hunk 7]
```diff
@@ -77,11 +76,12 @@ export const TransactionFlowSection = () => {
...
         />
         {recipientAddress && (
-          <ConfirmInfoRow
+          <ConfirmInfoAlertRow
+            alertKey={RowAlertKey.FirstTimeInteraction}
             label={t('to')}
+            ownerId={transactionMeta.id}
             style={{
               flexDirection: FlexDirection.Column,
-              alignItems: AlignItems.flexStart,
```
Comment (OGPoyraz) [inline ui/pages/confirmations/components/confirm/info/token-transfer/transaction-flow-section.tsx:84]: I've tested this to be sure, this is not needed.

**Review by digiwand — APPROVED** [review]

**Review by matthewwalsh0 — APPROVED** [review]

**Pull request merged**